Quadriga/src/main/java/edu/asu/spring/quadriga/dto/TextFileDTO.java

```diff
@@ -0,0 +1,82 @@
+package edu.asu.spring.quadriga.dto;
+
+import java.io.Serializable;
+
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * @author Nischal Samji
+ * 
+ *         DTO for Text File Operations
+ *
+ */
+@Entity
+@Table(name = "tbl_textfiles")
+@XmlRootElement
+public class TextFileDTO implements Serializable {
+
+    private static final long serialVersionUID = -1798070786993154676L;
+
+    @Id
+    @Basic(optional = false)
```

Comment:
don't put constraints like that here. Those checks should be done before in the business logic layer.